latch/type_engine/__init__.py

```diff
@@ -0,0 +1,393 @@
+    )
+
+
+def best_effort_python_val(t: typing.T):
+    """Produce a "best-effort" default value given a python type.
+
+    Unlike `guess_python_val` there is no Literal value acting as a guide.
+    """
+
+    if t in _primitive_table:
+        return _primitive_table[t]
+
+    if t is list:
+        return []
+
+    file_like_table = {
+        LatchDir: LatchDir("latch:///foobar"),
+        LatchFile: LatchFile("latch:///foobar"),
+    }
+    if t in file_like_table:
+        return file_like_table[t]
+
+    if type(t) is enum.EnumMeta:
+        return list(t.__members__.values())[0]
+
+    if get_origin(t) is list:
+        list_args = get_args(t)
+        if len(list_args) == 0:
+            return []
+        return [best_effort_python_val(arg) for arg in list_args]
+
+    if get_origin(t) is typing.Union:
+        return best_effort_python_val(get_args(t)[0])
+
+    if get_origin(t) is None:
+
+        if "__dataclass_fields__" in t.__dict__:
+            fields = t.__dict__["__dataclass_fields__"]
+            dataclass_args = {
+                k: best_effort_python_val(v.type) for k, v in fields.items()
+            }
+            return t(**dataclass_args)
+
+        # TODO : record implementation
+
+    raise NotImplementedError(
+        f"Unable to produce a best-effort value for the python type {t}"
+    )
+
+
+def guess_python_val(literal: _Literal, python_type: typing.T):
+    """Transform flyte literal value to native python value."""
+
+    if literal.scalar is not None:
+        if literal.scalar.none_type is not None:
+            return None
+
+        if literal.scalar.primitive is not None:
+            primitive = literal.scalar.primitive
+
+            if primitive.string_value is not None:
+                if type(python_type) is enum.EnumMeta:
+                    return list(python_type.__members__.values())[0]
+                return str(primitive.string_value)
+
+            if primitive.integer is not None:
+                return int(primitive.integer)
+            if primitive.float_value is not None:
+                return float(primitive.float_value)
+            if primitive.boolean is not None:
+                return bool(primitive.boolean)
+
+        if literal.scalar.blob is not None:
+            blob = literal.scalar.blob
+            dim = blob.metadata.type.dimensionality
+            if dim == 0:
+                return LatchFile(blob.uri)
+            else:
+                return LatchDir(blob.uri)
+
+    # collection
+    if literal.collection is not None:
+        p_list = []
+        for item in literal.collection.literals:
+            p_list.append(guess_python_val(item, get_args(python_type)[0]))
+        return p_list
+
+    # sum: TODO
+
+    # enum: TODO
+
+    # class: TODO
+
+    raise NotImplementedError(
+        f"The flyte literal {literal} cannot be transformed to a python type."
+    )
+
+
+def build_python_literal(python_val: any, python_type: typing.T) -> (str, str):
```

Comment:
```suggestion
def build_python_literal(python_val: object, python_type: typing.Type[object]) -> tuple[str, str]:
```
the `tuple` thing has to be like that otherwise pyright complains

I'm pretty sure you straight up do not have pyright enabled lol

Reply:
kinda weird function name too